### PR TITLE
Added directive location documentation.

### PR DESCRIPTION
**Pull request by PascalSenn**
Adds directive locations to the docs

### CODE REVIEW

**Review by rstaib — COMMENTED** [review]
As said before the middleware part will be rewritten. Just tell me what you're missing and I will put it on my list. Also the the writing style should be active throughout the entire document.

File: website/src/docs/hotchocolate/defining-a-schema/directives.md [hunk 1]
```diff
@@ -304,3 +304,130 @@ If there were more directives in the query, they would be appended to the direct
 So, now the order would be like the following: `a, b, c, d, e, f`.
 
 Every middleware can execute the original resolver function by calling `ResolveAsync()` on the `IDirectiveContext`.
+
+# Directive Locations
+
+There are two categories of directives. Type system directives and executable directives.
```
[the remark this answers is not in the document]
Comment (PascalSenn) [reply website/src/docs/hotchocolate/defining-a-schema/directives.md]: remove the duplicated parts and made it active

**Comment by tobias-tengler** [comment]
@PascalSenn  I really like this change - I think we should definitely include it!
I made some minor adjustments, let me know if they work for you. :)

**Review by michaelstaib — APPROVED** [review]

**Comment by tobias-tengler** [comment]
Can we re-target this to develop?

**Comment by PascalSenn** [comment]
Replaced by #4127

**Pull request closed without merging**